File: AchtungLunneSpa/ServerCode/BoundingBox.cs
using TypeLite;

namespace SPATest.ServerCode
{
    [TsClass]
    public class BoundingBox
    {
        public Vector2D TopLeft { get; set; }
        public Vector2D BottomRight { get; set; }

        public BoundingBox(Vector2D topLeft, Vector2D bottomRight)
        {
            TopLeft = topLeft;
            BottomRight = bottomRight;
        }

        public bool isInBounds(Vector2D position)
        {
            return position.X >= TopLeft.X &&
                position.X <= BottomRight.X &&
                position.Y >= TopLeft.Y &&
                position.Y <= BottomRight.Y;
        }
    }
}

File: AchtungLunneSpa/Scripts/LunnEngine/BoundingBox.ts
class BoundingBox implements SPATest.ServerCode.BoundingBox {

    topLeft: SPATest.ServerCode.Vector2D;
    bottomRight: SPATest.ServerCode.Vector2D;

    constructor(topLeft: SPATest.ServerCode.Vector2D, bottomRight: SPATest.ServerCode.Vector2D) {
        this.topLeft = topLeft;
        this.bottomRight = bottomRight;
    }

    isInBounds(position: SPATest.ServerCode.Vector2D) {
        return position.x >= this.topLeft.x &&
            position.x <= this.bottomRight.x &&
            position.y >= this.topLeft.y &&
            position.y <= this.bottomRight.y;
    }

}
export = BoundingBox;